orte/util/proc_info.c

```diff
@@ -164,19 +168,29 @@ int orte_proc_info(void)
     /* get the process id */
     orte_process_info.pid = getpid();
 
-    /* get the nodename */
-    gethostname(hostname, sizeof(hostname));
+    /* get the nodename (retrieves from a global variable) */
+    hostname = opal_gethostname();
+    /* copy into a local version so that we can manipulate it */
+    hostname_len = strlen(hostname);
+    if( hostname_len > 0 ) {
+        local_hostname = malloc( strlen(hostname) );
+    }
+    if( NULL == local_hostname ) {
+       return OPAL_ERR_BUFFER; 
+    }
+    strcpy(local_hostname, hostname);
```

Comment:
You could probably replace the above block with
```c
local_hostname = strdup(opal_gethostname());
if (NULL == local_hostname) {
    return OPAL_ERR_BUFFER;
}

Reply:
Thanks, fixed.